server/controllers/userEventController.ts

```diff
@@ -0,0 +1,162 @@
+import { StatusCodes } from "http-status-codes";
+import { Request, Response } from "express";
+import { DatabaseUserEvent } from "../services/database.user.event";
+
+import UnauthorizedError from "../errors/UnauthorizedError";
+import mongoose, { ObjectId } from "mongoose";
+import BadRequestError from "../errors/BadRequest";
+import { UserEventResponse } from "../types/responses/userEventResponse";
+import {
+	isValidPartialUserEventRequest,
+	isValidUserEventRequest
+} from "../utils/isValidRequest";
+import { includeIfDefined } from "../utils/includeIfDefined";
+
+export const create = async (req: Request, res: Response) => {
+	if (!req.session.userid) {
+		throw new UnauthorizedError();
```

Comment:
this should be in the auth middleware. Request shouldn't reach this point unless client is authorized.